cmd/containerd-shim-runhcs-v1/service.go

```diff
@@ -432,15 +432,8 @@ func (s *service) DiagStacks(ctx context.Context, req *shimdiag.StacksRequest) (
 		span.AddAttributes(trace.StringAttribute("pod-id", s.tid))
 	}
 
-	buf := make([]byte, 4096)
-	for {
-		buf = buf[:runtime.Stack(buf, true)]
-		if len(buf) < cap(buf) {
-			break
-		}
-		buf = make([]byte, 2*len(buf))
-	}
```

Comment:
Why did we roll our own (if you know haha) that's essentially the exact same besides the buffer that's four times larger? `Stack` has had that implementation since 1.6 I believe